content/browser/cosmetic_filters_communication_impl.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/browser/cosmetic_filters_communication_impl.h"
+
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "base/values.h"
+#include "brave/content/browser/cosmetic_filters_observer.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/render_process_host.h"
+
+namespace content {
+
+// static
+void CosmeticFiltersCommunicationImpl::CreateInstance(
+    content::RenderFrameHost* render_frame_host,
+    CosmeticFiltersObserver* cosmetic_filters_observer) {
+  if (!render_frame_host->cosmetic_filters_communication_impl_) {
```

Comment:
I'm not following why we're attaching this to the render frame host like this. It seems to add a lot of patching and there doesn't seem to be any reason for it that I can figure out.